basyx.submodelservice/basyx.submodelservice-backend-inmemory/src/main/java/org/eclipse/digitaltwin/basyx/submodelservice/InMemorySubmodelService.java

```diff
@@ -271,6 +271,10 @@ public void setFileValue(String idShortPath, String fileName, InputStream inputS
 		String uniqueFileName = createUniqueFileName(idShortPath, fileName);
 
 		FileMetadata fileMetadata = new FileMetadata(uniqueFileName, fileSmElement.getContentType(), inputStream);
+		
+		if(fileRepository.exists(fileMetadata.getFileName())) {
```

Comment:
What is the purpose of this?

Reply:
If someone tries to save a file, that already exists, an FileHandlingException is thrown in the save function of the FileRepository. If this happens in the MongoDB part (in any repsoitory), the only way to solve this I found was to delete the existing file and save the new. If this is the wrong thinking and there is a better way, I am willing to change it.